libtenzir/builtins/operators/show.cpp

```
@@ -0,0 +1,451 @@
+                  && builder.add(synopsis.synopsis->max_import_time)
+                  && builder.add(synopsis.synopsis->version)
+                  && builder.add(synopsis.synopsis->schema.name()))) {
+            diagnostic::error("failed to add partition entry")
+              .note("from `show {}`", args_.aspect.inner)
+              .emit(ctrl.diagnostics());
+            co_return;
+          }
+        }
+        co_yield builder.finish();
+      } else if (args_.aspect.inner == "types") {
+        auto types = type_set{};
+        auto error = caf::error{};
+        ctrl.self()
+          .request(catalog, caf::infinite, atom::get_v, atom::type_v)
+          .await(
+            [&types](type_set& result) {
+              types = std::move(result);
+            },
+            [&error](caf::error err) {
+              error = std::move(err);
+            });
+        co_yield {};
+        if (error) {
+          ctrl.abort(std::move(error));
+          co_return;
+        }
+        // TODO: this would be the preferred way to render the data, but
+        // to_definition() does not provide a uniform schema.
+        /*
+        for (const auto& type : types) {
+          auto definition = type.to_definition();
+          auto builder = adaptive_table_slice_builder{type::infer(definition)};
+          auto row = builder.push_row();
+          auto field = row.push_field("type");
+          add(field, definition);
+          co_yield builder.finish();
+        }
+        */
+        auto builder = adaptive_table_slice_builder{type_type()};
+        for (const auto& type : types) {
+          auto row = builder.push_row();
+          auto err = row.push_field("name").add(type.name());
+          auto structure = row.push_field("structure").push_record();
+          auto f = detail::overload{
+            [&](const auto&) {
+              auto err
+                = structure.push_field("basic").add(fmt::to_string(type));
+              (void)err;
```

Uh, discarding errors. Perhaps you can assert?